### PR TITLE
[bug][issue-2477] login failed when ldap password was changed

### LINKED ISSUE #2477: [Bug] login failed when ldap password was changed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### Java Version

11

### Scala Version

2.12.x

### StreamPark Version

2.1.0

### Flink Version

1.15.3

### deploy mode

yarn-application

### What happened

 login failed when ldap password was changed

### Error Exception

_No response_

### Screenshots

ldap_error

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoulii**
## What changes were proposed in this pull request

Issue Number: close #2477 

## Verifying this change

- *Manually verified the change by testing locally.*

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / **no**)

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -155,6 +155,18 @@ public void updatePassword(User userParam) {
...
+  @Override
+  @Transactional(rollbackFor = Exception.class)
+  public void updateSaltPassword(User userParam) {
+    User user = getById(userParam.getUserId());
+    ApiAlertException.throwIfNull(user, "User is null. Update password failed.");
+    user.setSalt(userParam.getSalt());
+    user.setPassword(userParam.getPassword());
+    LambdaQueryWrapper<User> queryWrapper =
+        new LambdaQueryWrapper<User>().eq(User::getUserId, user.getUserId());
+    this.baseMapper.update(user, queryWrapper);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java]: We can make the code more concise:
```
    this.baseMapper.updateById(user);
```

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -155,6 +155,18 @@ public void updatePassword(User userParam) {
     this.baseMapper.update(user, queryWrapper);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java]: this.baseMapper.updateById(user)..

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java (hunk 1 above)
Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java]: nice suggestion, done

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java]: done

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**